### ISSUE TITLE
Nightly full-pipeline canary with report archiving

### ISSUE THREAD

**Issue by niznik-dev** [niznik-dev/cruijff_kit#389]
## Summary

Run the complete scaffold → train (1 epoch) → eval → report pipeline nightly and archive the reports. Over time, this creates a time series of reports on a fixed experiment, letting us detect drift in model behavior, scoring changes, or subtle regressions that per-component tests miss.

## Approach

Create a standalone script (e.g., `tests/integration/gpu/nightly_canary.sh`) that:
1. Scaffolds the capitalization workflow test (1B, rank4 only, 1 epoch)
2. Runs fine-tuning via **direct subprocess** (no SLURM — GPUs sit idle at night)
3. Runs inspect eval on the fine-tuned model
4. Runs the report/analysis generation
5. Archives the report with a date stamp

## Report Archiving

- **Local:** `~/nightly-canary-reports/` on della-rse (persistent, long-term history)
- **GitHub:** Upload as workflow artifact (30-day retention, visible from GitHub UI)
- Optional: simple diff script that compares today's report to yesterday's and flags significant metric changes

## Workflow

- File: `.github/workflows/nightly-canary.yml`
- Cron: daily (e.g., 3:00 AM UTC, after torchtune and inspect smoke tests)
- Runner: `[self-hosted, della-cruijff-gpu]`
- Runtime target: ~12-15 min

## Constraint

No Claude Code on the runner — everything must be scripted.

## Dependencies

Depends on patterns established in the nightly torchtune and inspect smoke tests.

## Relationship to #363

This is the scriptable version of what #363 envisions. #363 can remain open for skill-level testing and the "tracking log" aspects.

## Effort

Medium-large (~half day). Most complexity is in the standalone script.

## Related

Part of nightly GPU runner test strategy. See also #362, #363.

**Comment by niznik-dev** [comment]
Folding in scope from #363 (closing that one as a duplicate). Two ideas worth keeping from its body:

1. **Variant rotation** — instead of pinning the canary to a single `capitalization 1B rank4 1ep` config, rotate through the three workflow test variants (`workflow_test.md`, `workflow_test_base.md`, `workflow_test_recipe.md`) across the week. Covers more surface area at the same per-night cost.

2. **Structured per-run log** alongside the report archive:

   ```yaml
   - date: 2026-03-13
     variant: "LoRA Comparison"   # or "Base vs Fine-tuned" / "Recipe Defaults"
     branch: main
     commit: abc1234
     result: pass                  # pass / fail / partial
     duration: ~12m
     notes: "Clean run, no issues"
     failure_stage: null           # design / scaffold / run / eval (when fail)
     failure_detail: null
   ```

   Easier to grep / dashboard than reading individual reports. Storage options: in-repo markdown, machine-parseable YAML/JSON, or wiki.

—MxC

**Comment by niznik-dev** [comment]
<img src="https://raw.githubusercontent.com/niznik-dev/llm-personas/main/fusions/mattie/mattie-icon.png" width="36" align="absmiddle"> 🎻 **MxC** ✳️

Closing as part of the September 2026 archive triage.

This design lives on the `della-cruijff-gpu` self-hosted runner, which goes away with the archive, and GitHub disables scheduled Actions on archived repos regardless. The drift-detection idea (a fixed experiment rerun nightly, reports archived as a time series) is still a good one for anyone with a GPU runner; the body above is the complete spec. Recorded in the handoff notes alongside the other nightly GPU workflows that will be dead on arrival.

—MxC
